data/awsrds/v0/main.go

```diff
@@ -0,0 +1,448 @@
+				"status": {Kind: &structpb.Value_StringValue{StringValue: "Successfully executed SQL statement"}}}}
+		case taskQuery:
+			query := input.GetFields()["query"].GetStringValue()
+
+			err := QueryUser(client, query)
+			if err != nil {
+				return nil, err
+			}
+
+			output = &structpb.Struct{Fields: map[string]*structpb.Value{
+				"status": {Kind: &structpb.Value_StringValue{StringValue: "Successfully executed SQL statement"}}}}
+		default:
+			return nil, fmt.Errorf("task %s not supported", e.Task)
+		}
+		outputs = append(outputs, output)
+	}
+	return outputs, nil
+}
+
+func UpsertUser(client *sqlx.DB, tableName *string, data *structpb.Struct) error {
+	// Begin constructing SQL statement
+	sqlStatement := "INSERT INTO " + *tableName + " ("
+	var columns []string
+	var placeholders []string
+	var updateColumns []string
+	values := make(map[string]any)
+
+	for dataKey, dataValue := range data.GetFields() {
+		columns = append(columns, dataKey)
+		placeholders = append(placeholders, ":"+dataKey)
+		updateColumns = append(updateColumns, fmt.Sprintf("%s = VALUES(%s)", dataKey, dataKey))
+		values[dataKey] = dataValue.AsInterface()
+	}
+
+	sqlStatement += strings.Join(columns, ", ") + ") VALUES (" + strings.Join(placeholders, ", ") + ") "
+	sqlStatement += "ON DUPLICATE KEY UPDATE " + strings.Join(updateColumns, ", ")
+
+	// Prepare and execute the statement using NamedExec
+	_, err := client.NamedExec(sqlStatement, values)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func SelectUser(client *sqlx.DB, tableName *string, criteria *structpb.Struct, from int, to int) (*structpb.ListValue, error) {
+	// Begin constructing SQL statement
+	sqlStatement := "SELECT "
+	var where []string
+	var columns []string
+
+	for criteriaKey, criteriaValue := range criteria.GetFields() {
+		if criteriaValue.GetKind() != nil {
+			switch criteriaValue.Kind.(type) {
+			case *structpb.Value_StringValue:
+				// If the value is a string, quote it in the SQL statement
+				where = append(where, fmt.Sprintf("%s = '%s'", criteriaKey, criteriaValue.GetStringValue()))
+			case *structpb.Value_NumberValue, *structpb.Value_BoolValue:
+				// If the value is a number or bool, use it directly without quotes
+				where = append(where, fmt.Sprintf("%s = %v", criteriaKey, criteriaValue.AsInterface()))
+			}
+		}
+
+		columns = append(columns, criteriaKey)
+	}
+
+	var notAll string
+	if to == 0 && from == 0 {
+		notAll = ""
+	} else {
+		notAll = " LIMIT " + strconv.Itoa(to-from+1) + " OFFSET " + strconv.Itoa(from-1)
+	}
+
+	sqlStatement += strings.Join(columns, ", ")
+
+	sqlStatement += " FROM " + *tableName
+	if len(where) > 0 {
+		sqlStatement += " WHERE " + strings.Join(where, " AND ")
+	}
+	sqlStatement += notAll
+
+	// Prepare and execute the statement
+	rows, err := client.Queryx(sqlStatement)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	// Prepare the result ListValue
+	result := &structpb.ListValue{}
```

Comment:
Same as comment above.
If you use the component struct, you can have the benefit of making code cleaner.

[Example PR](https://github.com/instill-ai/component/pull/190/files#diff-e489ebc6b09b1110ffef7a388d25d6199476e1ec1cfc19c60db73e39735bd98dL205)